src/main/java/com/pinterest/secor/common/KafkaClient.java

```diff
@@ -164,9 +164,12 @@ private SimpleConsumer createConsumer(String host, int port, String clientName)
         return new SimpleConsumer(host, port, 100000, 64 * 1024, clientName);
     }
 
-    public SimpleConsumer createConsumer(TopicPartition topicPartition) {
+    private SimpleConsumer createConsumer(TopicPartition topicPartition) {
         HostAndPort leader = findLeader(topicPartition);
-        LOG.debug("leader for topic {} partition {} is {}", topicPartition.getTopic(), topicPartition.getPartition(), leader.toString());
+        LOG.debug("leader for topic {} partition {} is {}", topicPartition.getTopic(), topicPartition.getPartition(), leader);
+        if (leader == null) {
+            return null;
```

Comment:
Can you add a warning logging here?

Reply:
Sure. The calling code warns too, but if you don't think that's too many warnings, I'll add it.